### ISSUE TITLE
Malformed crashdump causes OOM by allocating far too much memory

### ISSUE THREAD

**Issue by 5225225** [luser/rust-minidump#381]
On the base64 decoded data of `TURNUJOnAAAA/2ZmZFlmZmZmZkAKCmZwCrv///8K/wo=`, it OOMs when parsing through the `minidump::Minidump::read(data)` method, and when running `minidump-stackwalk` (version 0.9.6), I get

```
# minidump-stackwalk crash.txt          
memory allocation of 45097156624 bytes failed
zsh: abort (core dumped)  minidump-stackwalk crash.txt
```

**Comment by Swatinem** [comment]
I think this is related to https://github.com/luser/rust-minidump/issues/326.
Also ❤️ @5225225 I also saw your https://github.com/getsentry/symbolic/issues/476 which shows a very similar issue.

**Comment by 5225225** [comment]
Yeah, both this crate and symbolic seems incredibly underfuzzed (as in, not at all) considering what they're doing.

I wouldn't expect any RCE or anything, it's Rust, not `$memory_unsafe_language`, but still, it would be nice if whatever processes crashes doesn't crash itself :D
